src/System.Private.CoreLib/src/System/AppContext.Unix.cs

```diff
@@ -17,7 +19,34 @@ public static string BaseDirectory
         {
             get
             {
-                throw new NotImplementedException();
+                int currentSize = Path.MaxPath;
+                for (;;)
+                {
+                    char[] buffer = ArrayPool<char>.Shared.Rent(currentSize);
+
+                    // Get full path to the executable image
+                    int actualSize = Interop.Sys.GetExecutableAbsolutePath(buffer, buffer.Length);
+
+                    if (actualSize < 0)
+                    {
+                        // The call to GetExecutableAbsolutePath function failed.
+                        ArrayPool<char>.Shared.Return(buffer);
+                        // TODO: should we throw an exception here?
+                        return "";
+                    }
+
+                    Debug.Assert(actualSize > 0);
+                    if (actualSize <= buffer.Length)
+                    {
+                        string fileName = new string(buffer, 0, actualSize);
+                        ArrayPool<char>.Shared.Return(buffer);
+
+                        return fileName.Substring(0, fileName.LastIndexOf(Path.DirectorySeparatorChar));
```

Comment:
The path should contain terminating slash. I do not think it will with this impl. corefx tests are checking for it (https://github.com/dotnet/corefx/issues/18196).